### ISSUE TITLE
Matrix > SuperTable > Matrix failing

### ISSUE THREAD

**Issue by Emkaytoo** [engram-design/SuperTable#137]
### Description
I think that Matrix > SuperTable > Matrix has issues, but I'm not sure how to check it.

### Steps to reproduce

1. Manually configured Matrix field, added block, added SuperTable. No issues.
2. Added a child-matrix field to SuperTable field above, attempted to save.
3. Got the error below. Also, this error prevents me from loading any page in the url string (`/admin/settings/fields/2`). I have to revert to an backup database to undo the change.
4. This led to the error described in #134 

### Additional info

```
ParseError

syntax error, unexpected '$', expecting function (T_FUNCTION) or const (T_CONST)

1. in {project}/storage/runtime/compiled_classes/ContentBehavior.php at line 91
828384858687888990919293949596979899100                            'video' => true,
    ];
 
    // Properties
    // =========================================================================
 
    /**
     * @var mixed Value for field with the handle “”.
     */
    public $;
 
    /**
     * @var mixed Value for field with the handle “altText”.
     */
    public $altText;
 
    /**
     * @var mixed Value for field with the handle “articleBuilder”.
     */
```
- Plugin version: 2.0.4
- Craft version: 3.0.0 RC15

**Comment by tylerbrostrom** [comment]
Much to my horror, I came across this error as well. Might this be expected-ish behavior? Expected-ish meaning that SuperTable wasn't meant to be a _literal_ solution for matrix-in-matrix. Also, I’m on Craft 3 and haven't used SuperTable for Craft 2. So 🤷‍♂️.

I hope that's not the case. I'm attempting to use SuperTable as a carousel builder inside of a content builder, and I think it's a pretty compelling use-case.

So that’s:

- `contentBuilder` (Matrix)
    - `...` 
    - `carouselBlock` (MatrixBlock)
        - `...` 
        - `carousel` (Static SuperTable)
            - `slides` (Matrix)
                - `videoSlide` (MatrixBlock)
                    - `video` (Dukt Video)
                - `imageSlide` (MatrixBlock)
                    - `image` (Asset)

Which is a way nicer authoring experience than this:

- `contentBuilder` (Matrix)
    - `...` 
    - `carouselBlock` (MatrixBlock)
        - `...` 
        - `carousel` (SuperTable)
            - (SuperTable Row) x n
                - `video` (Dukt Video)
                - `image` (Asset)

Which requires a field description of "Hey, could you, like, fill one or the other but not, like, fill both of them?".

@Emkaytoo, I'm curious if you're getting this error having set the SuperTable to `static`, or if it happens no matter what.

**Comment by Emkaytoo** [comment]
@bro-strummer I don't believe my SuperTable was set to, but I haven't tested it out.

Also to your point, this setup is not ideal. For a builder as complex as the ones we're making, Neo would have been the best way to do that. I'm not sure what the timetable is for Neo in Craft 3 though.

**Comment by Emkaytoo** [comment]
I can confirm that it still fails with a static SuperTable field.

**More information**: I went step by step, and it succeeds when adding a Matrix field, but fails after adding blocks to the matrix field.

**Comment by engram-design** [comment]
This has been fixed in [2.0.5](https://github.com/verbb/super-table/releases/tag/2.0.5).

**Comment by ghost** [comment]
Just encountered exactly the same issue as well. I am on Supertable 2.0.7
My last database backup is a few days old. Didn't expect a change to a field to fuck my database up :/

EDIT: Turns out the DB wasn't faulty. I had to fix a few weird lines in the `storage/runtime/compiled_classes/ContentBehaviour.php` and now everything works as expected.
